### ISSUE TITLE
Ending Meet fails

### ISSUE THREAD

**Issue by dproyt** [SHUR1K-N/MeetNinja-Google-Meet-Bot#6]
yo bro the bot joins the classes and leaves too but when it leaves the class it encounters some error and bot stops although it has left the class but it says leaving meet failed any fixes and there is no sound

**Comment by SHUR1K-N** [comment]
Uncomment [line #255](https://github.com/SHUR1K-N/MeetNinja-Google-Meet-Bot/blob/24b92d39c9072c963cfdefe8f039216c31702f1d/MeetNinja.py#L255) from MeetNinja.py:

`# print(e)`

Then, reproduce the failure (set Meet duration to 0.1 to make it end the call in ~10secs). This time, the failure will show itself in the form of a traceback (error message) in the MeetNinja console window. You may then share this traceback here.

**Comment by dproyt** [comment]
![hiii](https://user-images.githubusercontent.com/74549153/99349644-82e8e680-28c2-11eb-9df7-2bedb0ad4053.PNG)

**Comment by SHUR1K-N** [comment]
You need to _uncomment_ the mentioned line by changing it from `# print(e)` to just `print(e)`. This will then show the traceback after the ″Failed!″ message.

**Comment by dproyt** [comment]
iill just give the code to u see if there r any probs\

**Comment by dproyt** [comment]
[Uploading MeetNinja.zip…]()

**Comment by dproyt** [comment]
[MeetNinja.zip](https://github.com/SHUR1K-N/MeetNinja-Google-Meet-Bot/files/5551449/MeetNinja.zip)

**Comment by SHUR1K-N** [comment]
Referring to the following section of your version of the code:
```
MEETS = {
                "1 https://meet.google.com/bbr-oqhz-cgg?authuser=0&hs=179": "2020 11 17 10 47 15"
                "2 https://meet.google.com/txf-umme-hxo?authuser=0&hs=179"  "2020 11 17 11 47 48"
                "3 http://meet.google.com/aen-jzdp-ttq?authuser=0&hs=179"  "2020 11 17 12 48 11"
                "4 https://meet.google.com/uhy-rgxt-edo?authuser=1&hs=179" " 2020 11 17 15 1 18"
}
```
You have missed the colons (:) in all Meets except the first one, and all commas (,) separating the Meets. Since "MEETS" is a Python dictionary, it is supposed consist of key-value pairs denoted by the _colon_ (:), and separated from the other key-value pairs using a _comma_ (,), both of which you are omitting. Hence, the failure.

**Try adding colons and commas as below:**
```
MEETS = {
                "1 https://meet.google.com/bbr-oqhz-cgg": "2020 11 17 10 47 15",
                "2 https://meet.google.com/txf-umme-hxo": "2020 11 17 11 47 48",
                "3 http://meet.google.com/aen-jzdp-ttq": "2020 11 17 12 48 11",
                "4 https://meet.google.com/uhy-rgxt-edo": "2020 11 17 15 1 18",
}
```
![Dictionary](https://user-images.githubusercontent.com/42811989/99354855-36ef6f00-28cd-11eb-9a18-0fe08b7bb2ee.png)

Also, in your version, [line #143](https://github.com/SHUR1K-N/MeetNinja-Google-Meet-Bot/blob/24b92d39c9072c963cfdefe8f039216c31702f1d/MeetNinja.py#L143) has the `time.sleep()` value as `1`, which is too low. There is a reason the value is `3` in the original code; the sign-in procedure needs to fully complete before the next step (navigating to the first Meet URL) takes place.

**Comment by dproyt** [comment]
putting a colon gives a syntax error

**Comment by SHUR1K-N** [comment]
The previous comment has been edited/updated, kindly check; **you are missing the commas as well**. Making the above mentioned fixes in the code works on my side.

**Comment by dproyt** [comment]
ok ill checck

**Comment by dproyt** [comment]
ok wht about the leave meeeting error  how to solve?

**Comment by SHUR1K-N** [comment]
Again, refer to the detailed comment regarding the colons and commas you are missing in your code.

**Comment by dproyt** [comment]
could u check the program if u could find any error about leave class after that ill correct it

**Comment by SHUR1K-N** [comment]
Yes, and **I have already mentioned it** in the **detailed comment regarding the colons and commas you are missing in your code**.

**Comment by dproyt** [comment]
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\lenovo\Downloads\MeetNinja-Google-Meet-Bot-master\MeetNinja-Google-Meet-Bot-master\MeetNinja.py", line 257, in <module>
    print(e)
NameError: name 'e' is not defined
PS C:\Users\lenovo\Downloads\MeetNinja-Google-Meet-Bot-master\MeetNinja-Google-Meet-Bot-master

**Comment by SHUR1K-N** [comment]
I will need the traceback from _before_ the "During handling of the above exception, another exception occurred:" line.

**Comment by dproyt** [comment]
Entering Google Meet #1...Traceback (most recent call last):
  File "c:\Users\lenovo\Downloads\MeetNinja-Google-Meet-Bot-master\MeetNinja-Google-Meet-Bot-master\MeetNinja.py", line 238, in <module>
    attendMeet()
  File "c:\Users\lenovo\Downloads\MeetNinja-Google-Meet-Bot-master\MeetNinja-Google-Meet-Bot-master\MeetNinja.py", line 154, in attendMeet
    joinButton = wait.until(when.element_to_be_clickable((by.XPATH, joinButtonPath)))
  File "C:\Users\lenovo\AppData\Local\Programs\Python\Python38\lib\site-packages\selenium\webdriver\support\wait.py", line 80, in until
    raise TimeoutException(message, screen, stacktrace)
selenium.common.exceptions.TimeoutException: Message: 


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\lenovo\Downloads\MeetNinja-Google-Meet-Bot-master\MeetNinja-Google-Meet-Bot-master\MeetNinja.py", line 257, in <module>
    print(e)
NameError: name 'e' is not defined
PS C:\Users\lenovo\Downloads\MeetNinja-Google-Meet-Bot-master\MeetNinja-Google-Meet-Bot-master>

**Comment by SHUR1K-N** [comment]
Can you share a screenshot of the Chrome window after this error has occurred?

**Comment by dproyt** [comment]
CHROME CLOSES AFTER THIS ERROR

**Comment by SHUR1K-N** [comment]
That is because you have removed the last `genericError()` line in the `except:` block (starting at [line #254](https://github.com/SHUR1K-N/MeetNinja-Google-Meet-Bot/blob/24b92d39c9072c963cfdefe8f039216c31702f1d/MeetNinja.py#L254)). This is how it should look:
```
    except:
        print(e)
        genericError()
```

**Comment by dproyt** [comment]
YEP IFIGURED IT OU AND NOW ITS WORKS LIKE BUTTER THANKS FOR UR HELP BROTHER

**Comment by dproyt** [comment]
BRO ANY CODE TO UNBLOCK SPEAKERS

**Comment by SHUR1K-N** [comment]
Can you be more specific as to what you are wanting to achieve/do? I did not quite get what you mean.

**Comment by dproyt** [comment]
ACTUALLY I CANT HEAR ANYTHING IN MEET  WHEN I GO TO SETTINGS IT SHOWS SPEAKERS ARE BLOCKED

**Comment by SHUR1K-N** [comment]
You may open a separate Issue thread for this specific issue.
